index.js

```diff
@@ -66,6 +67,12 @@ class Fetcher {
             throw error;
         }
     }
+
+    _setDefaultOptions(options) {
+        const defaultOptions = options || {};
+        defaultOptions.timeout = defaultOptions.timeout || constants.defaultRequestTimeout;
+        return defaultOptions;
```

Comment:
By using the following code we will be able to pass `0` for timeout to disable it.

```suggestion
        const mergedOptions = Object.assign({}, constants.DefaultRequestOptions, options);
        
        return mergedOptions;
```